src/lib.rs

```diff
@@ -21,27 +78,36 @@ fn hash(msgs: Vec<Vec<u8>>) -> Vec<String> {
     keys
 }
 
-/// Verify signatures for an array of messages.
+/// Verify signatures for an array of messages (includes HMAC key support).
+///
+/// Takes an HMAC key as the first argument and an array of messages as the second argument.
+/// The HMAC key must be of type `string` or `ArrayBuffer`. Message signatures are verified without
+/// an HMAC key if the value of the argument is a `string` with value `none`.
 ///
-/// Takes an array of messages as the only argument. If verification fails, the cause of the error
-/// is returned along with the offending message. Note: this method only verifies message signatures;
-/// it does not perform full message validation (use `verify_validate_message_array` for complete
-/// verification and validation).
+/// If verification fails, the cause of the error is returned along with the offending message.
+/// Note: this method only verifies message signatures; it does not perform full message validation
+/// (use `verify_validate_message_array` for complete verification and validation).
 #[node_bindgen(name = "verifySignatures")]
-fn verify_messages(array: Vec<String>) -> (Option<String>, Option<Vec<String>>) {
+fn verify_messages(hmac_key: HmacKey, array: Vec<String>) -> (Option<String>, Option<Vec<String>>) {
```

Comment:
Could you share some thoughts on the reasons for hmac_key as the 1st argument? I'm actually fine with it as an API, I'm just looking at line 105 (calling `par_verify_message_values`) which puts hmac on the 2nd arg and thinking what was the reason behind those.

Reply:
I'm not sure why the Sunrise Choir chose to put hmac as the second argument in `ssb-verify-signatures`. I chose to put it as the first argument here to keep some level of consistency with the current implementation of `ssb-validate` (`(state, hmac_key, msg)` for `append()` and `appendKVT()`).

Reply:
Oh, I see! Thanks. That's where the inconsistency is. I think hmac-first is okay.